src/elements/content-sidebar/ActivitySidebar.js

```diff
@@ -717,7 +738,34 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
             );
         }
 
-        this.setState({ feedItems, activityFeedError: undefined });
+        const baseChanges = { feedItems, activityFeedError: undefined };
+
+        // If activeFeedEntryId is not set or
+        // feed items are already in state (we want to load the active entry (reply) data only if the feed items are loaded for the first time) or
+        // replies should not be loaded at all or
+        // active feed entry type is other than comment,
+        // set feedItems state without any additional actions
+        if (
+            !activeFeedEntryId ||
+            feedItemsState ||
+            !shouldShowReplies ||
+            activeFeedEntryType !== FEED_ITEM_TYPE_COMMENT
+        ) {
+            this.setState({ ...baseChanges });
+            return;
+        }
+
+        this.getActiveFeedEntryData(feedItems)
+            .then(({ id, type }) => {
+                this.setState({
+                    ...baseChanges,
```

Comment:
I'm a bit concerned about race conditions here. What happens if `fetchFeedItems` is called again while this promise is still in progress? Would we wind up with stale feed items? It seems like we'd want to handle this conditional promise chain prior to the success callback being invoked?

Reply:
These additional asynchronous actions (fetching reply data and its parent replies) will take place only once when feed items are not yet loaded. From what I can tell, during that time fetchFeedItems will not be called as the whole Activity Sidebar is in the "loading state".

I've also moved that additional actions logic outside of `fetchFeedItemsSuccessCallback`.